docs/proposals/locking.md

```diff
@@ -108,3 +108,188 @@ support:
 |9  |Cater for read-only binary files when merging locally<ul><li>Because files are read-only this might prevent merge from working when actually it's valid.</li><li>Always fine to merge the latest version of a binary file to anywhere else</li><li>Fine to merge the non-latest version if user is aware that this may cause merge problems (see Breaking the rules)</li><li>Therefore this feature is about dealing with the read-only flag and issuing a warning if not the latest</li></ul>|
 |10 |List current locks<ul><li>That the current user has</li><li>That anyone has</li><li>Potentially scoped to folder</li></ul>|`git lfs lock --list [paths...]`
 |11 |Reject a push containing a binary file currently locked by someone else|pre-receive hook on server, allow --force to override (i.e. existing parameter to git push)
+
+## Implementation details
+### Types
+To make the implementing locking on the lfs-test-server as well as other servers
+in the future easier, it makes sense to create a `lfs/lock` package that can be
+depended upon from any server. This will go along with Steve's refactor which
+touches the `lfs` package quite a bit.
```

Comment:
I think the idea is that `github.com/github/git-lfs/lfs` is _not_ a package that we want 3rd parties to use. This should be extracted to a top level package like `github.com/github/git-lfs/lock`. May make sense to call it `github.com/github/git-lfs/lfslock` so that it's obvious when callers use it:

```go
lfslock.Lock{Path: "img/foo.psd", ...}
```

I dont'e care about the name so much, but I do want it as a top level package under the repo.